services/app-api/src/postgres/findUser.ts

```diff
@@ -1,17 +1,33 @@
-import { PrismaClient, User } from '@prisma/client'
+import { PrismaClient } from '@prisma/client'
 import { convertPrismaErrorToStoreError, StoreError } from './storeError'
+import {
+    AdminUserType,
+    CMSUserType,
+    StateUserType,
+    UserType,
+} from '../domain-models'
 
 export async function findUser(
     client: PrismaClient,
     id: string
-): Promise<User | StoreError> {
+): Promise<UserType | StoreError> {
     try {
         const findResult = await client.user.findUnique({
             where: {
                 id: id,
             },
         })
-        return findResult ?? ({} as User)
+
+        switch (findResult?.role) {
+            case 'ADMIN_USER':
+                return findResult as AdminUserType
```

Comment:
it would be nice if we didn’t have to use as here, and since this is repeated code in each of the handlers maybe we can have a more strict converter.